src/mds/Server.cc

```diff
@@ -7273,6 +7369,26 @@ void Server::handle_client_getvxattr(const MDRequestRef& mdr)
       // since we only handle ceph vxattrs here
       r = -ENODATA; // no such attribute
     }
+  } else if (xattr_name == "ceph.dir.subvolume.snaps.visible"sv) {
+    if (!cur->is_dir()) {
+      r = -ENOTDIR;
+    } else {
+      MutationImpl::LockOpVec lov;
+      lov.add_rdlock(&cur->snaplock);
+      if (!mds->locker->acquire_locks(mdr, lov)) {
+        dout(20) << "handle_client_getvxattr could not acquire rdlock on "
+                 << *cur << dendl;
+        return;
+      }
+      const auto srnode = cur->get_projected_srnode();
+      if (!(srnode && srnode->is_subvolume())) {
+        r = -EINVAL;
+      } else {
+        *css << srnode->is_snapdir_visible();
+      }
+      dout(20) << "dropping rdlock on " << *cur << dendl;
+      mds->locker->drop_locks(mdr.get());
```

Comment:
This would not have been necessary. Locks are dropped via request finish.

Reply:
removed the entire block of snaplock acquisition as per https://github.com/ceph/ceph/pull/64345#discussion_r2282649721